src/control/common/pci_utils.go

```diff
@@ -37,3 +42,289 @@ func ParsePCIAddress(addr string) (dom, bus, dev, fun uint64, err error) {
 
 	return
 }
+
+// PCIAddress represents a standard PCI address with domain and BDF.
+type PCIAddress struct {
+	Domain   string
+	Bus      string
+	Device   string
+	Function string
+}
+
+func (pa *PCIAddress) String() string {
+	if pa == nil {
+		return ""
+	}
+
+	return fmt.Sprintf("%s:%s:%s.%s", pa.Domain, pa.Bus, pa.Device, pa.Function)
+}
+
+// Equals compares string representation of address.
+func (pa *PCIAddress) Equals(o *PCIAddress) bool {
+	if pa == nil || o == nil {
+		return false
+	}
+
+	return o.String() == pa.String()
+}
+
+// IsVMDBackingAddress indicates whether PCI address is a VMD backing device.
+func (pa *PCIAddress) IsVMDBackingAddress() bool {
+	if pa == nil {
+		return false
+	}
+
+	return pa.Domain != "0000"
+}
+
+// NewPCIAddress creates a PCIAddress stuct from input string.
+func NewPCIAddress(addr string) (*PCIAddress, error) {
+	dom, bus, dev, fun, err := parsePCIAddress(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	pa := &PCIAddress{
+		Bus:      fmt.Sprintf("%02x", bus),
+		Device:   fmt.Sprintf("%02x", dev),
+		Function: fmt.Sprintf("%01x", fun),
+	}
+
+	if dom == 0 {
+		pa.Domain = "0000"
+	} else {
+		pa.Domain = fmt.Sprintf("%x", dom)
+	}
+
+	return pa, nil
+}
+
+// PCIAddressList represents a list of PCI addresses.
+type PCIAddressList []*PCIAddress
```

Comment:
Given that it doesn't seem like we're really using the aliased slice functionality (i.e. we've got `Add()` and `Len()` methods instead of just using `append()` and `len()` directly), it might more sense to make this a struct with a map holding the addresses so that you don't need to dedupe all the time -- the map will take care of that for you.

Usually it makes sense to use a type alias for a slice when you just need a few convenience methods (e.g. `String()`).